src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToFloatWarningLevels.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using FluentAssertions;
+
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToFloatWarningLevels : SdkTest
+    {
+        private const string targetFrameworkNet6 = "net6.0";
+        private const string targetFrameworkNet7 = "net7.0";
+        private const string targetFrameworkNetFramework472 = "net472";
+
+        public GivenThatWeWantToFloatWarningLevels(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [InlineData(targetFrameworkNet6, 6)]
+        [InlineData(targetFrameworkNet7, 7)]
+        [InlineData(targetFrameworkNetFramework472, 4)]
+        [RequiresMSBuildVersionTheory("16.8")]
+        public void It_defaults_WarningLevel_To_The_Current_TFM_When_Net(string tfm, int warningLevel)
+        {
+            var testProject = new TestProject
+            {
+                Name = "HelloWorld",
+                TargetFrameworks = tfm,
+                IsExe = true,
+                SourceFiles =
+                {
+                    ["Program.cs"] = @"
+                        using System;
+
+                        namespace ConsoleCore
+                        {
+                            class Program
+                            {
+                                static void Main()
+                                {
+                                }
+                            }
+                        }
+                    ",
+                }
+            };
+
+            var testAsset = _testAssetsManager
+                .CreateTestProject(testProject, identifier: "warningLevelConsoleApp"+tfm, targetExtension: ".csproj");
+
+            var buildCommand = new GetValuesCommand(
+                Log,
+                Path.Combine(testAsset.TestRoot, testProject.Name),
+                tfm, "WarningLevel")
```

Comment:
Only one property/Item, but it can get the values of a property or the items of an itemgroup with metadata. We do a build and inject a target that computes this [here](https://github.com/dotnet/sdk/blob/5d2faf1bf54d4aa3fb81537c1359d1df2f6a0b8a/src/Tests/Microsoft.NET.TestFramework/Commands/GetValuesCommand.cs#L64-L126)